### DISCUSSION TITLE
how to display dm_only commands in guild channel help

### DISCUSSION THREAD

**Discussion by dbuteau** [Rapptz/discord.py#6397] [Q&A]
Hello!
i've defined some commands to only be called from private message to my bot, because it's private/personnal commands (rgpd concern).

But i fear people don't think to try sending help message in private to get list of this commands.
i would like than this commands display in the result of help command in guild chat.

i've tried arguments like show_hidden=true or hidden=false, but nothing work. Is it possible? if yes how?

example of my code (cog file):
```python
class Privacy(commands.Cog):

    @commands.group(pass_context=True)
    @commands.dm_only(hidden=False)
    async def privacy(self, ctx):
        """ Commands to control sensitives data about you (send command as private message to the bot) """"
        if ctx.invoked_subcommand is None:
            await self.bot.say('Invalid sub command passed...')

   @privacy.command()
   @commands.dm_only(hidden=False):
   async def delete(self, ctx):
       """ delete all infos the bot know about you """
       [...]

   @privacy.command()
   @commands.dm_only(hidden=False):
   async def infos(self, ctx):
       """ display all info the bot know about you """
       [...]
```

**Comment by Zomatree** [comment]
https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.HelpCommand.verify_checks

**Reply by dbuteau** [reply]
Thank you for your quick reply. 

Maybe i didn't understood correctly the doc. 
It say:
> verify_checks: specifies if commands should have their Command.checks called and verified. Defaults to True

and
> Commands.Checks: A list of predicates that verifies if the command could be executed with the given Context

In my mind than mean than if i use `verify_checks=False` it will no more check if the command is called as private_message or not.

it's like i didn't use `@commands.dm_only()` at all, no? i didn't understood correctly?

**Reply by Zomatree** [reply]
dm_only is a check, the help command loops thought command.checks and see if they pass or fail, by default it doesnt show any commands where the checks fail, if the help command is ran outside a dm the check will fail and thus the command will not be shown.

**Comment by Rapptz** [comment]
The default help command doesn't show commands that the user can't run, so those commands won't be shown in a guild context. This is controlled by `verify_checks` (which is `True` by default).

Note that passing `hidden=True` to the `dm_only` check is invalid.

**Comment by dbuteau** [comment]
Thank you for your answers.
If i understood well, if i want to use the verify_check flag, i need to redo the entire helpcommand and no more use the default help command.

It's, at my eyes, little overkill just for my use. 
I've by-passed this by deleting the `@commands.dm_only()` line and managing channel check in the method:
```
@commands.group(pass_context=True)
async def privacy(self, ctx):
    if isinstance(ctx.channel, discord.channel.DMChannel):
        [...]
    else:
        await ctx.message.delete()
        await ctx.author.send('You should send privacy commands in private message')
```
The job is done, quickly, simply, all the privacy group and subcommands are dm_only

**Reply by Rapptz** [reply]
No. As was mentioned, `verify_checks=True` is the default. You do not have to change anything. You run the help command in a guild and it doesn't show those commands. Unrolling the check into the body of the command does the complete opposite and won't hide the command at all.

**Reply by dbuteau** [reply]
> Unrolling the check into the body of the command does the complete opposite and won't hide the command at all.

Edit: after rereading...
yes it was exactly what i wanted in fact.
1. displaying dm_only commands when wrote help on channel (bc lambda users don't think than writing `help` in dm will give them another result and miss lot of features)
2. but keeping dm_only check just to catch and return error when this commands are called on channel

For example what i wanted help command return on channel:
```
Public commands:
  top10   display top 10 games played by members
  up      saying to other you'r up for playing. when more than 1 person is up will suggest to you games you have in common
Private commands:  only private messages with bot
  import steam <your_steam_id>  import your owned steam games into our databases
  import gog <your_gog_id>  import your owned gog games into our databases
  untrackme                say to not watch your discord activity for games you are playing
  forgetme                  delete all we know about you in our database
```
